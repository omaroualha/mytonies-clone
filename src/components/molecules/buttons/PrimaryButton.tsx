import { Headline } from "../../atoms";
import React, { FC } from "react";
import { PressableProps, TouchableOpacity } from "react-native";
import styled, { css } from "styled-components/native";
import { flex, FlexProps, variant } from "styled-system";

interface PrimaryButtonProps extends PressableProps, FlexProps {
  label: string;
  size?: "L" | "M" | "S";
  variant?: "DEFAULT" | "DARK" | "LIGHT";
  disabled?: boolean;
}

const PrimaryButtonBackground = styled(TouchableOpacity)<PrimaryButtonProps>`
  flex-direction: row;
  justify-content: center;
  align-items: center;
  min-width: 100px;
  ${({ theme }) => {
    return variant({
      prop: "size",
      variants: {
        L: {
          height: 50,
          borderRadius: "30",
          paddingHorizontal: theme.space.M,
        },
        M: {
          height: 40,
          borderRadius: "20",
          paddingHorizontal: theme.space.M,
        },
        S: {
          height: 30,
          borderRadius: "10",
          paddingHorizontal: theme.space.S,
        },
      },
    });
  }}
  ${({ theme, disabled }) => {
    if (disabled) {
      return css`
        background-color: ${theme.colors.neutral.grey20};
      `;
    }
    return variant({
      variants: {
        DEFAULT: {
          backgroundColor: theme.colors.primary.brand,
        },
        DARK: {
          backgroundColor: theme.colors.primary.dark,
        },
        LIGHT: {
          backgroundColor: "#b9d29b",
        },
      },
    });
  }}
    
    ${flex}
`;

export const PrimaryButton: FC<PrimaryButtonProps> = (props) => {
  const { size, label } = props;
  return (
    <PrimaryButtonBackground activeOpacity={0.6} {...props}>
      <Headline
        variant={size == "L" ? "H3" : "H4"}
        color="primary.dark"
        textAlign={"center"}
      >
        {label}
      </Headline>
    </PrimaryButtonBackground>
  );
};
PrimaryButton.defaultProps = { size: "L", variant: "DEFAULT" };
