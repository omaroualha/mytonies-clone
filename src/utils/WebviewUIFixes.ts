export const uiFixesApp = (): string => {
  const styles = `
    .main-navigation{
      display: none;
    }
    .browserupgrade .SiteTheme__Site-sc-1ycpvs-0 kKWbES .sc-ehmTmK fstLMM SiteTheme__StyledMainNavigation-sc-1ycpvs-1 cdoBQQ {
      display: none;
    }
    .b-account-page-breadcrumb-wrapper {
      display: none;
    }
    `;
  const minifiedStyles = styles.replace(/\s\s+/g, "");

  return `;(function() {
      if (document.body.dataset.injectedFromApp) return;
      document.head.insertAdjacentHTML("beforeend", "<style>${minifiedStyles}</style>");
      document.body.dataset.injectedFromApp = 'true';
    })();
    `;
};
