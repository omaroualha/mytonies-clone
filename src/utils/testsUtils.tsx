import * as React from "react";
import { BaseTheme } from "@/theme/BaseTheme";
import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

/** Render helper that renders `ui` within `NavigationContainer`. */
const renderWithNavigation = (screen: JSX.Element) => {
  return render(
    <ThemeProvider theme={BaseTheme}>
      <NavigationContainer>{screen}</NavigationContainer>
    </ThemeProvider>
  );
};

const renderWithTheme = (component: JSX.Element) => {
  return render(<ThemeProvider theme={BaseTheme}>{component}</ThemeProvider>);
};

export const testsUtils = { renderWithNavigation, renderWithTheme };
