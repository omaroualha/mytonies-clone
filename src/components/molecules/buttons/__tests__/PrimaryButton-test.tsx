import * as React from "react";
import { screen } from "@testing-library/react-native";
import { PrimaryButton } from "../PrimaryButton";
import { testsUtils } from "@/utils/testsUtils";

describe(`PRIMARY BUTTON`, () => {
  it(`Primary button DEFAULT variant`, () => {
    testsUtils.renderWithTheme(<PrimaryButton label="button" />);
    expect(screen.getByText("button")).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it(`Primary button BRAND variant`, () => {
    testsUtils.renderWithTheme(
      <PrimaryButton variant="BRAND" label="button" />
    );
    expect(screen.getByText("button")).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
