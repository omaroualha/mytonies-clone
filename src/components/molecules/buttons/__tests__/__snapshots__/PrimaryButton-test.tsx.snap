// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRIMARY BUTTON Primary button BRAND variant 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#c12822",
      "borderRadius": 8,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 8,
      "minHeight": 35,
      "opacity": 1,
      "padding": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "Museo-300",
          "fontSize": 15,
          "letterSpacing": 0,
          "lineHeight": 18,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    variant="P2"
  >
    button
  </Text>
</View>
`;

exports[`PRIMARY BUTTON Primary button DEFAULT variant 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e5e5e5",
      "borderRadius": 8,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 8,
      "minHeight": 35,
      "opacity": 1,
      "padding": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "Museo-300",
          "fontSize": 15,
          "letterSpacing": 0,
          "lineHeight": 18,
        },
        Object {
          "color": "#222222",
        },
      ]
    }
    variant="P2"
  >
    button
  </Text>
</View>
`;
